src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -0,0 +1,911 @@
+          case PredicateCondition::Equals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? 1.0f / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::NotEquals: {
+            const auto total_distinct_count =
+                std::max(scan_statistics_object->total_distinct_count(), HistogramCountType{1.0f});
+            selectivity = total_distinct_count > 0 ? (total_distinct_count - 1.0f) / total_distinct_count : 0.0f;
+          } break;
+
+          case PredicateCondition::LessThan:
+          case PredicateCondition::LessThanEquals:
+          case PredicateCondition::GreaterThan:
+          case PredicateCondition::GreaterThanEquals:
+          case PredicateCondition::BetweenInclusive:
+          case PredicateCondition::BetweenExclusive:
+          case PredicateCondition::BetweenLowerExclusive:
+          case PredicateCondition::BetweenUpperExclusive:
+          case PredicateCondition::In:
+          case PredicateCondition::NotIn:
+          case PredicateCondition::Like:
+          case PredicateCondition::NotLike:
+            // Lacking better options, assume a "magic" selectivity for >, >=, <, <=, ... Any number would be equally
+            // right and wrong here. In some examples, this seemed like a good guess ¯\_(ツ)_/¯
+            selectivity = 0.5f;
+            break;
+
+          case PredicateCondition::IsNull:
+          case PredicateCondition::IsNotNull:
+            Fail("IS (NOT) NULL predicates should not have a 'value' parameter.");
+        }
+
+      } else {
+        /**
+         * Estimate ColumnVsValue
+         */
+
+        Assert(predicate.value.type() == typeid(AllTypeVariant), "Expected AllTypeVariant");
+
+        // TODO(anybody) For (NOT) LIKE predicates that start with a wildcard, Histograms won't yield reasonable
+        //               results. Assume a magic selectivity for now
+        if (predicate.predicate_condition == PredicateCondition::Like) {
+          // Lacking better options, assume a "magic" selectivity for LIKE. Any number would be equally
+          // right and wrong here. In some examples, this seemed like a good guess ¯\_(ツ)_/¯
+          selectivity = 0.1f;
+          return;
+        }
+        if (predicate.predicate_condition == PredicateCondition::NotLike) {
+          // Lacking better options, assume a "magic" selectivity for NOT LIKE. Any number would be equally
+          // right and wrong here. In some examples, this seemed like a good guess ¯\_(ツ)_/¯
+          selectivity = 0.9f;
+          return;
+        }
+
+        auto value2_variant = std::optional<AllTypeVariant>{};
+        if (predicate.value2) {
+          if (predicate.value2->type() != typeid(AllTypeVariant)) {
+            selectivity = 1.0f;
+            return;
```

Comment:
Sometimes you return for `selectivity = ...`, sometimes you don't. Can we make this consistent?

Reply:
Now returning for all cases where `scan_statistics_object->sliced()` should not be called.